### PR TITLE
Issue 2444 gui testcases

### PR DESCRIPTION
**Pull request by maryvictol**
This PR provides update NATGateway testcases and tests and includes 
- #2444 
- routes grouping added for routes with the same names and protocols

### CODE REVIEW

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java [hunk 1]
```diff
@@ -140,15 +160,21 @@ public NATGatewayAO checkCreationScheduled(final String ipAddressOrServerName, f
         return this;
     }
 
+    public NATGatewayAO expandGroup(final String serverName, final String port) {
+        sleep(3, SECONDS);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java:163]: Do we actually need a sleep here?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java [hunk 2]
```diff
@@ -132,6 +150,8 @@ public NATGatewayAO checkNoRouteRecord(final String ipAddressOrServerName, final
     }
 
     public NATGatewayAO checkCreationScheduled(final String ipAddressOrServerName, final String port) {
+        sleep(3, SECONDS);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java:152]: Do we actually need a sleep here?

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java [hunk 3]
```diff
@@ -314,11 +356,10 @@ public NATGatewayAO addRoute() {
              return parent();
          }
 
-         public NATAddRouteAO addMorePorts(final String port) {
-             context().findAll(byText("Port"))
-                     .filter(hidden).first()
-                     .closest(".dd-route-modal__form-item-container")
-                     .find(".dd-route-modal__form-item").find("input")
+         public NATAddRouteAO addMorePorts(final String port, int portNumber) {
+             context().findAll(By.className("cp-nat-route-port-control")).get(portNumber-1)
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java]: Add spaces between `portNumber-1`, please

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_8.md [hunk 4]
```diff
@@ -15,11 +16,12 @@ Test verifies that it's possible to add route with resolved IP to the existing r
...
 | 9 | Clear the ***Port*** field and input `443` | |
-| 10 | Click **Add** button | |
-| 11 | Click **Save** button | Icon with status ***CREATION SCHEDULED*** appears for record added at steps 10 |
-| 12 | Wait 1 minute and click **Refresh** button | The record added at step 10 has <li> icon with status ***ACTIVE*** <li> ***Internal IP*** that is the same as the value stored at step 5 |
-| 13 | At the **Runs** page, click the run launched at step 16 of [2232_2](2232_2.md) case| |
-| 14 | Click the **SSH** hyperlink | |
-| 15 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl <Server_name_2>:443 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to <Server_name_2> <internal_ip> port <port>`, <br> where `<internal_ip>` is *Internal IP* stored at step 5, `<port>` is *port* specified at step 9 |
+| 10 | Select `UDP` value in the ***Protocol*** field | |
+| 11 | Click **Add** button | |
+| 12 | Click **Save** button | Icon with status ***CREATION SCHEDULED*** appears for record added at steps 10 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_8.md]: Fix to `... added at step 11`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- it's possible to add new route with several ports and the same Protocol
+- confirmation dialog appears if user tries to leave page with unsaved changes
+
+**Prerequisites**:
+- Admin user
+- Perform [2232_2](2232_2.md) case
+- Test server `<Server_name_3>`.
+
+**Preparations**
+1. Open the **Settings** page. Click the **SYSTEM MANAGEMENT** tab
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: Add `Click the **NAT GATEWAY** tab`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+| 11 | Click **+** icon near the `<Server_name_3>` name to expand group | 2 new records with the name specified at step 6 and ports specified at steps 7 and 9 are shown in the External Resources list |
+| 12 | Try to switch to the **Logs** tab | Confirmation message `You have unsaved changes. Continue?` appears |
+| 13 | Click **No** button on the confirmation pop-up | |
+| 14 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 10 |
+| 15 | Wait 1 minute and click **Refresh** button | The records added at step 10 have <li> icons with status ***ACTIVE*** <li> the same *Internal IPs* and different *Internal Ports* |
+| 16 | Store *Internal IP* for the route added at step 10 | |
+| 17 | At the **Runs** page, click the run launched at step 16 of [2232_2](2232_2.md) case| |
+| 18 | Click the **SSH** hyperlink | |
+| 19 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl <Server_name_3>:80 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to <Server_name_3> <internal_ip> port 80`, <br> where `<internal_ip>` is *Internal IP* stored at step 15 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: > ... stored at step 15

Fix to `... stored at step 16`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+| 20 | In the opened tab, enter and perform the command: <br> `curl <Server_name_3>:443 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to <Server_name_3> <internal_ip> port 443`, <br> where `<internal_ip>` is *Internal IP* stored at step 15 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: > ... stored at step 15

Fix to `... stored at step 16`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md [hunk 8]
```diff
@@ -25,12 +25,13 @@ Test verifies that it's possible to add several route records with the same serv
...
+| 13 | Wait 1 minute and click **Refresh** button | Grouping row for `<Server_name_3>` appears that contains <li> **+** icon near the server name to expand group <li> list of Ports `80, 443`
+| 14 | Click **+** icon near the `<Server_name_3>` name to expand group | The records added at steps 10 and 11 are shown in the group list and have <li> icons with status ***ACTIVE*** <li> the same *Internal IPs*, different *Internal Ports* and *Comments* |
+| 15 | Store *Internal IP* for the route saved at step 12 | |
+| 16 | At the **Runs** page, click the run launched at step 16 of [2232_2](2232_2.md) case| |
+| 17 | Click the **SSH** hyperlink | |
+| 18 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl <Server_name_4>:80 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to <Server_name_4> <internal_ip> port 80`, <br> where `<internal_ip>` is *Internal IP* stored at step 14 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md]: > ... stored at step 14

Fix to `... stored at step 15`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md [hunk 9]
```diff
@@ -25,12 +25,13 @@ Test verifies that it's possible to add several route records with the same serv
...
+| 19 | In the opened tab, enter and perform the command: <br> `curl <Server_name_4>:443 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to <Server_name_4> <internal_ip> port 443`, <br> where `<internal_ip>` is *Internal IP* stored at step 14 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md]: > ... stored at step 14

Fix to `... stored at step 15`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md [hunk 10]
```diff
@@ -22,12 +22,11 @@ Test verifies that
...
-| 9 | Input `443` into the 2nd ***Port:*** field | **Add** button becomes active |
-| 10 | Click **Add** button | 2 new records with the name specified at step 6 and ports specified at steps 7 and 9 appear in the External Resources list |
-| 11 | Try to switch to the **Logs** tab | Confirmation message `You have unsaved changes. Continue?` appears |
-| 12 | Click **No** button on the confirmation pop-up | |
-| 13 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 10 |
+| 8 | Click **Add Port** button | |
+| 9 | Input `443` into the 2nd ***Port:*** field | |
+| 10 | Select `UDP` value in the 2nd ***Protocol*** field | |
+| 11 | Click **Add** button | 2 new records with the name specified at step 6 and ports specified at steps 7 and 9 appear in the External Resources list |
+| 12 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 10 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: > ... added at steps 10

Fix to `... added at step 11`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md [hunk 11]
```diff
@@ -22,12 +22,11 @@ Test verifies that
...
 | 14 | Wait 1 minute and click **Refresh** button | The records added at step 10 have <li> icons with status ***ACTIVE*** <li> the same *Internal IPs* and different *Internal Ports* |
 | 15 | Store *Internal IP* for the route added at step 10 | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: > ... added at step0

Fix to `... added at step 11`

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md [hunk 12]
```diff
@@ -22,12 +22,11 @@ Test verifies that
...
-| 10 | Click **Add** button | 2 new records with the name specified at step 6 and ports specified at steps 7 and 9 appear in the External Resources list |
-| 11 | Try to switch to the **Logs** tab | Confirmation message `You have unsaved changes. Continue?` appears |
-| 12 | Click **No** button on the confirmation pop-up | |
-| 13 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 10 |
+| 8 | Click **Add Port** button | |
+| 9 | Input `443` into the 2nd ***Port:*** field | |
+| 10 | Select `UDP` value in the 2nd ***Protocol*** field | |
+| 11 | Click **Add** button | 2 new records with the name specified at step 6 and ports specified at steps 7 and 9 appear in the External Resources list |
+| 12 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 10 |
 | 14 | Wait 1 minute and click **Refresh** button | The records added at step 10 have <li> icons with status ***ACTIVE*** <li> the same *Internal IPs* and different *Internal Ports* |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: Fix numeration for items 14-19 (they shall be 13-18 now)

**Review by NShaforostov — APPROVED** [review]
Fix minor notes before merging please (see comments for details).

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md (hunk 12 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md (hunk 11 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md (hunk 10 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2444_1.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md (hunk 9 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_10.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md (hunk 7 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_9.md]: Done

File: docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_8.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232-2444_NAT_Gateway/2232_8.md]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java (hunk 3 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NATGatewayAO.java]: Done

**Review by kamyshova — APPROVED** [review]

**Pull request merged**